doc/design.md

```diff
@@ -0,0 +1,93 @@
+# Kubernetes MAC pool address allocation
+
+## Motivation
+
+As of date, there is no way to ensure MAC address collisions do not occur within a Kubernetes cluster. Many CNI plugins allocate MAC addresses randomly, with no assurance of collision prevention.
+
+Even if a CNI plugin ensures it never creates collisions, a cluster operating with multiple CNI plugins or sharing a broadcast domain with another cluster will remain collision-prone.
+
+## Limitations
+CNI plugins that do not implement CNI's MAC address specification will not be supported. The cluster must have multus installed and configured.
+
+## Goals
+
+This document proposes a collision-free allocation of MAC addresses within a single cluster.
+
+### Allocation
+This specification will describe an intermediate step prior to the initialization of Pods within a kubernetes cluster. During this step, a Pod's template will be read, processed, and modified to specify a MAC address.
+
+It is also to be noted that besides Pods, the proposed behavior will be generic enough to support future entities that may require MAC address allocation. For convenience sake, I will only refer to Pods throughout this document.
+
+### Collision prevention
+This specification will describe how the allocation process (as described in 3.1) will ensure no collisions may occur.
```

Comment:
there is no "3.1", please mention the section by its name

Reply:
done